doc/architecture/class_diagram.puml

```diff
@@ -0,0 +1,73 @@
+@startuml class_diagram
+
+' Setting ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+' Components '''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+class Main {
+  + main(): Ret.CODE
+  - add_parser(): argparse.ArgumentParser
+}
+
+class Ret {
```

Comment:
Its not necessary to add a class, as the enumeration itself in python is a class. You can go with "enum Ret ..."